src/gui/transferlistwidget.cpp

```diff
@@ -607,6 +607,57 @@ void TransferListWidget::askNewCategoryForSelection()
     } while(invalid);
 }
 
+void TransferListWidget::askAddTagsForSelection()
+{
+    const QStringList tags = askTagsForSelection("Add Tags");
+    foreach (const QString &tag, tags)
+        addSelectionTag(tag);
+}
+
+void TransferListWidget::askRemoveTagsForSelection()
+{
+    const QStringList tags = askTagsForSelection("Remove Tags");
+    foreach (const QString &tag, tags)
+        removeSelectionTag(tag);
+}
+
+void TransferListWidget::confirmRemoveAllTagsForSelection()
+{
+    QMessageBox::StandardButton response;
+    response = QMessageBox::question(
+        this, tr("Remove All Tags"), tr("Remove all tags from selected torrents?"),
+        QMessageBox::Yes | QMessageBox::No);
+    if (response == QMessageBox::Yes)
+        clearSelectionTags();
+}
+
+QStringList TransferListWidget::askTagsForSelection(const QString &dialog_title)
+{
+    QStringList tags;
+    bool invalid;
+    do {
+        bool ok;
+        invalid = false;
+        const QString tags_input = AutoExpandableDialog::getText(
+            this, tr(dialog_title.toStdString().c_str()), tr("Comma-separated tags:"),
```

Comment:
I moved the tr()'s to the call sites. There are actually only two possible values for the title.